Label example that demonstrates sizing to texture_size and wrapping with text_size.

arlowhite opened the pull request:
Creates a new example: *examples/widgets/label_texture_size.py*

I spent some time looking through Kivy's examples and never found one that really helped me understand how texture_size and text_size are used. So, I created this example while working on some documentation #3448.

A few issues:
- [x] Is this a good file name and location? Maybe *label_sizing.py* would be better?
- [x] max_lines does not behave as I would expect. Is this a problem with my example or max_lines itself?
- [x] I was thinking of creating a button that outlines the Label.textures, would this be useful or just overcomplicate the example, which is already a bit complicated.

@matham This is just the example from our discussion on #3429

arlowhite commented on examples/widgets/label_texture_size.py:
```diff
@@ -0,0 +1,182 @@
+        text: 'Reset'
+        padding: mm(6), mm(3)
+        size: self.texture_size
+        # Align to right side and vertically align-middle with title
```
Here, I right-align and vertical-align the Reset button using property calculations and link to relevant doc. (Line length violates PEP8, but I think it's excusable for a URL)

- [ ] Should developers be encouraged to do this? Or would it be better to demonstrate using a FloatLayout with pos_hint as mentioned in the documentation?

arlowhite commented on examples/widgets/label_texture_size.py:
```diff
@@ -0,0 +1,182 @@
+        # Note: These calculations make the property dependencies explicit, see:
+        # http://kivy.org/docs/api-kivy.uix.widget.html#usage-of-widget-center-widget-right-and-widget-top
+        x: root.right - self.width - dp(4)
+        y: title.center_y - self.height / 2.0
```
Here, I right-align and vertical-align the Reset button using property calculations and link to relevant doc. (Line length violates PEP8, but I think it's excusable for a URL)

- [ ] Should developers be encouraged to do this? Or would it be better to demonstrate using a FloatLayout with pos_hint as mentioned in the documentation?

matham replied:
Either way is fine.

matham commented:
- Yeah, label_sizing is probably better. 
- Open a issue about the max_lines, I'd have to look into that.
- I don't know what you mean by "I was thinking of creating a button that outlines the Label.textures", can you give an example?

arlowhite commented:
Ok, renamed to label_sizing
I made the buttons modify all Labels, which is more instructive. Also, the text now has a newline so that you can see that halign does have an effect. (As you mentioned in the other PR)

Outlining the texture doesn't really demonstrate much; only the top left text has a texture not equal to the size, and that's already obvious.

- [ ] I can't figure out how to make the Spinners wider, setting width doesn't work. Maybe I should just use a Dropdown?

arlowhite commented:
Created a max_lines issue: #3450

arlowhite commented:
Figured out the Spinner problem, it inherits Button and I modified <Button> in KV.
Squashed everything to one commit. I'm done unless you have more comments.

matham commented:
:+1:
